source/extensions/health_checkers/thrift/client.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include "envoy/event/deferred_deletable.h"
+#include "envoy/network/connection.h"
+#include "envoy/upstream/upstream.h"
+
+#include "source/extensions/filters/network/thrift_proxy/thrift.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HealthCheckers {
+namespace ThriftHealthChecker {
+
+/**
+ * Outbound request callbacks.
+ */
+class ClientCallback : public Network::ConnectionCallbacks {
+public:
+  /**
+   * Called when the client needs a connection
+   */
+  virtual Upstream::Host::CreateConnectionData createConnection() PURE;
+
+  /**
+   * Called when a response is received.
+   * @param is_success indicate if the response is a success response
+   */
+  virtual void onResponseResult(bool is_success) PURE;
+};
+
+/**
+ * A single thrift client connection.
+ */
+class Client : public Event::DeferredDeletable {
+public:
+  /**
+   * Initialize the connection.
+   */
+  virtual void start() PURE;
+
+  /**
+   * Make a simple thrift request.
```

Comment:
Let's just go with: Sends the health check request.

Things could get less simple down the line :-)

Reply:
sure. i put "simple" since i expect this could be extend to different decoder, i.e., more usage than just health checker. but i agree "simple" is too vague. Let's go with your suggestion for now since hc is the only  use case